src/lib/address_bitset.cc

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+#include "address_bitset.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <unlikely.hpp>
+
+namespace ddprof {
+
+namespace {
+unsigned round_to_power_of_two(unsigned num) {
+  if (num == 0) {
+    return num;
+  }
+  // If max_addresses is already a power of two
+  if ((num & (num - 1)) == 0) {
+    return num;
+  }
+  // not a power of two
+  unsigned count = 0;
+  while (num) {
+    num >>= 1;
+    count++;
+  }
+  return 1 << count;
```

Comment:
This can be replaced by 
```
1 << (32 - _builtin_clz(x - 1));
```

Reply:
Thanks, good to know. I won't use the builtin as it is an init flow (so it does not really need the performance).